### PR TITLE
float values for raster legend

### PR DESCRIPTION
**Pull request by KaroFr**

### CODE REVIEW

**Comment by KaroFr** [comment]
[Legende verwendet falsche Werte bei Float](https://trello.com/c/BZyerfdy/197-legende-verwendet-falsche-werte-bei-float)

File: projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts [hunk 1]
```diff
@@ -47,19 +47,27 @@ export class RasterLegendComponent implements OnInit, OnChanges {
     }
 
     ngOnChanges(changes: SimpleChanges): void {
-        if (changes.symbology) {
-            const symbology = changes.symbology.currentValue;
+        if (changes.layer) {
+            const symbology = changes.layer.currentValue.symbology;
             this.numberPipeParameters = RasterLegendComponent.calculateNumberPipeParameters(symbology.colorizer.breakpoints);
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts:52]: I get an error message when using this branch:

> ERROR TypeError: breakpoints is undefined
    calculateNumberPipeParameters raster-legend.component.ts:58
    ngOnChanges raster-legend.component.ts:52
    Angular 36
    RxJS 5
    Angular 17
    RxJS 54
    changeProjectConfig project.service.ts:1006
    changeLayer project.service.ts:796
    RxJS 12
core.js:6479
    Angular 14
    RxJS 5
    Angular 17
    RxJS 54
    changeProjectConfig project.service.ts:1006
    changeLayer project.service.ts:796
    RxJS 23
    changeLayer project.service.ts:802
    toggleLegend project.service.ts:828
    toggleLegend layer-list.component.ts:132
    LayerListComponent_div_1_div_3_button_49_Template_button_click_0_listener layer-list.component.html:103
    Angular 23
    LayerListComponent_div_1_div_3_button_49_Template layer-list.component.html:103
    Angular 9
    LayerListComponent_div_1_div_3_Template layer-list.component.html:102
    Angular 6

Maybe this comes when you have a palette and call this method?

File: projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts [hunk 2]
```diff
@@ -47,19 +47,27 @@ export class RasterLegendComponent implements OnInit, OnChanges {
...
         }
     }
 
     private static calculateNumberPipeParameters(breakpoints: Array<ColorBreakpoint>): string {
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts:56]: In my opinion, this kind of utility method is very good testable and we should do this.
You can look for *.spec* files for other utilities for getting examples.
You can just try out different scenarios and validate that the outcome matches your expectations.

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]
I tried it out and it worked well. Nevertheless, I have two things addressed.

File: projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts (hunk 1 above)
Comment (KaroFr) [reply projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts]: It wasnt working for palette Colorizer bfd9bd28987dc5abefce0e66b801f16d51cefeca

File: projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts (hunk 2 above)
Comment (KaroFr) [reply projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts]: 4cbe85d84d5c95a010fe888a072f8c78d00d6fc0

**Review by ChristianBeilschmidt — APPROVED** [review]

**Comment by ChristianBeilschmidt** [comment]
bors merge

**Pull request merged**